app/redux/ducks/classify.spec.js

```diff
@@ -576,5 +576,164 @@ describe('Classifier actions', function () {
       })
       .then(done, done);
     });
+
+    describe('on error', function () {
+      describe('404 API errors', function () {
+        before(function () {
+          apiClient.type.restore();
```

Comment:
Non-blocking comment, but calling this here indicates to me that a test stub isn't being correctly restored elsewhere. Is this why this is here? If so, I think it'd be better to figure out what set of tests that don't have an after test block function defined to handle this. 

I prefer the use `after` or `afterEach` for cleaning up spies and stubs with `resetHistory` or `restore` and using `before` or `beforeEach` to setup for running tests.

Reply:
This is my terrible code. It’s overriding the mock API client that’s set up in the before block, which mocks the client for successful responses.